### PR TITLE
Backend/periodic tasks

### LINKED ISSUE #48: Periodic tasks wrapper
_The following is a snippet of a Slack-conversation in the #backend channel:_

So I have a question about running periodic tasks in the background:

Basically, I did some digging into Celery and alternatives; not going with the 'you should use Celery because everyone does', and here is what I found out:

1. Celery and similar tools are built to do _persistent_ tasks.
2. We actually do not need persistent tasks for your project (or at least the MVP), we only want tasks that repeatedly run at specific time intervals.

So instead, I went to look for something to build a simple time-based task-dispatcher with. It turns out that what we want exactly does not exist (or at least I was not able to find it), but that it can easily be implemented by using the `multiprocessing` (see https://docs.python.org/3.6/library/multiprocessing.html) built-in python library. Most importantly:

> multiprocessing is a package that supports spawning processes using an API similar to the threading module. The multiprocessing package offers both local and remote concurrency, effectively side-stepping the Global Interpreter Lock by using subprocesses instead of threads. Due to this, the multiprocessing module allows the programmer to fully leverage multiple processors on a given machine.

This would mean that background calls do not have an effect on the rest of the system, while not increasing app complexity (other solutions need some kind of intermediate queue store like Redis or RabbitMQ and run the task-spawner in a completely separate OS-process). I believe that Flask already uses multiprocessing under the hood to perform different web requests at the same time on multiple CPU cores.

If you guys agree, I'd like to create a very small piece of code that:
1. Spawns off a separate Process at startup, which calls itself once per second to see if it has tasks to run.
2. A function that adds periodic tasks to the queue (basically an equivalent of a crontab) of this object, possibly implemented as a function decorator ^^.

### PR DESCRIPTION
**Pull request by Qqwy**
Allows for the running of tasks in the background, as well as periodically.

This PR does not yet add tasks to the existing application, since that would be a separate feature to implement and the scope of this PR is already large enough as-is.

Fixes #48 .

### CODE REVIEW

File: pydash/pydash_app/impl/periodic_tasks.py [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+        return self.next_run_dt
+
+    def __repr__(self):
+        """
+        Overrides superclass to show `interval` and `next_run_dt` in the string representation.
+        """
+        return f"<{self.__class__.__name__} name={self.name}, target={self.target}, interval={self.interval}, next_run_dt={self.next_run_dt}>"
+
+
+class TaskScheduler:
```
Comment (FlyingBird95) [inline pydash/pydash_app/impl/periodic_tasks.py]: I would argue for moving this into a new file

File: pydash/pydash_app/impl/periodic_tasks.py [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+        except queue.Empty:
+            # Thrown by self._tasks_to_be_scheduled.get_nowait()
+            # when this cross-process queue happens to be empty.
+            pass
+
+    def _run_waiting_tasks(self, pool, current_time):
+        """
+        Runs all tasks whose `next_run_dt` has passed since last time.
+        """
+        # TODO restructure into for-comprehension?
```
Comment (FlyingBird95) [inline pydash/pydash_app/impl/periodic_tasks.py]: nice to use TODO-comments

File: pydash/pydash_app/impl/periodic_tasks.py (hunk 1 above)
Comment (InputUsername) [reply pydash/pydash_app/impl/periodic_tasks.py]: Agreed.

Comment (Qqwy) [reply pydash/pydash_app/impl/periodic_tasks.py]: Yes, that is indeed nicer. :+1:

**Pull request merged**